addons/account/models/account_move.py

```diff
@@ -2905,6 +2905,11 @@ def is_write_needed(line, values):
             }
             for move in container['records']
         }
+        currency_rates_values_before = {
+            move: move.invoice_currency_rate
+            for move in container['records']
+        }
```

Comment:
It looks ok except:
- you have to consider only invoices I guess
- what about this compute on account.move.line applying the rate line by line then:
```py
    @api.depends('currency_rate', 'balance')
    def _compute_amount_currency(self):
        for line in self:
            if line.amount_currency is False:
                line.amount_currency = line.currency_id.round(line.balance * line.currency_rate)
            if line.currency_id == line.company_id.currency_id and not line.move_id.is_invoice(True):
                line.amount_currency = line.balance
```